src/App.js

```diff
@@ -1,23 +1,20 @@
-import logo from './logo.svg';
-import './App.css';
+import { Route, Routes } from "react-router-dom";
+import Header from "./components/Header";
+// import "./App.css";
+import Home from "./pages/Home";
+import SingleArticle from "./pages/SingleArticle";
+import ArticleProvider from "./providers/ArticleProvider"
 
 function App() {
   return (
     <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <Header />
+      <Routes>
+        <Route path="/" element={<Home />} />
+        <Route path="/articles" element={<ArticleProvider />}>
+          <Route path="/articles/:article_id" element={<SingleArticle />} />
```

Comment:
Strictly speaking, this shouldnt be in this ticket, as its the ticket for all articles.